### ISSUE TITLE
Use multiple outputs in one loss function

### ISSUE THREAD

**Issue by HannahElisa** [keras-team/keras#13340]
Hello everybody,

I have a model producing as output a list of tensors with different shapes: outputs = [tensor1, tensor2, etc..]. I want to use all of this tensors in one single loss function to do some calculations. However, if I use the functional API and write
model = Model(input=inputs, output=outputs) 
my specified loss function will be computed on each tensor in the output list individually. 
Is there some way to apply the loss to the whole output list at once? 

Thanks for help and ideas!

**Comment by maxhardt** [comment]
There is! It just requires a small work around to work with the `y_true, y_pred` norm of the keras loss API.

Check out my response in #13453

**Comment by duf59** [comment]
You can use @maxhardt workaround, alternatively you can use `add_loss`, there is an example using the functional API [here](https://stackoverflow.com/questions/51680818/keras-custom-loss-as-a-function-of-multiple-outputs). I most often use the Model class API and do something like (pseudo-code):

```python
import tensorflow as tf

def custom_loss(out1, out2, y_true):
    """Your custom loss function."""
    loss = ...
    return loss


class MyModel(tf.keras.Model):

    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        # Define some layers
        self.out1 = tf.keras.layers.Dense(1)
        self.out2 = tf.keras.layers.Dense(1)

    def call(self, inputs):
        out1 = self.out1(inputs)
        out2 = self.out2(out1)
       
        # Add the loss to the model
        loss = custom_loss(out1=out1, out2=out2, y_true=y_true)
        self.add_loss(loss)

        return out2

model = MyModel().compile(optimizer='sgd')
```
